### PR TITLE
Extend avro reader to support more primitive types

### PR DESCRIPTION
**Pull request by yangy0000**
cherry-pick of https://github.com/trinodb/trino/commit/bf91d13aeab169a6882931d034b9d6dcab262984

Co-Authored-By: Elon Azoulay <elon.azoulay@gmail.com>

Test plan - (Please fill in how you tested your changes)

Please make sure your submission complies with our [Development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [Formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), and [Commit Message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests) guidelines. Don't forget to follow our [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution) for any code copied from other projects.

Fill in the release notes towards the bottom of the PR description.
See [Release Notes Guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) for details.

```
== RELEASE NOTES ==

Avro Decoder Changes
* Extend avro reader to support more primitive types 
```

### CODE REVIEW

File: presto-record-decoder/src/main/java/com/facebook/presto/decoder/avro/AvroColumnDecoder.java [hunk 1]
```diff
@@ -18,28 +18,36 @@
 import com.facebook.presto.common.type.BigintType;
 import com.facebook.presto.common.type.BooleanType;
 import com.facebook.presto.common.type.DoubleType;
+import com.facebook.presto.common.type.IntegerType;
```
Comment (yangy0000) [inline presto-record-decoder/src/main/java/com/facebook/presto/decoder/avro/AvroColumnDecoder.java:21]: @beinan  can you help to review the PR, thanks!

File: presto-record-decoder/src/main/java/com/facebook/presto/decoder/avro/AvroColumnDecoder.java [hunk 2]
```diff
@@ -257,12 +274,16 @@ private static void serializeGeneric(BlockBuilder blockBuilder, Object value, Ty
             case BOOLEAN:
                 type.writeBoolean(blockBuilder, (Boolean) value);
                 break;
+            case INTEGER:
             case BIGINT:
-                type.writeLong(blockBuilder, (Long) value);
+                type.writeLong(blockBuilder, ((Number) value).longValue());
```
Comment (beinan) [inline presto-record-decoder/src/main/java/com/facebook/presto/decoder/avro/AvroColumnDecoder.java:279]: Just wanna confirm, could the value be an Integer or something else here?

File: presto-record-decoder/src/main/java/com/facebook/presto/decoder/avro/AvroColumnDecoder.java [hunk 3]
```diff
@@ -189,13 +203,16 @@ private static Slice getSlice(Object value, Type type, String columnName)
     {
         switch (type.getTypeSignature().getBase()) {
             case VARCHAR:
-                if (value instanceof Utf8) {
+                if (type instanceof VarcharType && (value instanceof CharSequence || value instanceof GenericEnumSymbol)) {
```
Comment (beinan) [inline presto-record-decoder/src/main/java/com/facebook/presto/decoder/avro/AvroColumnDecoder.java:206]: do you have any context about why we get rid of Utf8 here? or is there any test covered the VARCHAR type to make sure we're not breaking anything?

**Review by beinan — COMMENTED** [review]
Thank you for this contribution! Looks good to me except a couple clarification questions.   Could you also update the release note section in this PR?  Thanks!

File: presto-record-decoder/src/main/java/com/facebook/presto/decoder/avro/AvroColumnDecoder.java (hunk 3 above)
Comment (mceste-volume) [reply presto-record-decoder/src/main/java/com/facebook/presto/decoder/avro/AvroColumnDecoder.java:206]: `Utf8` implements `CharSequence` so it should be more compatible in situations where the Avro schema used a different class for `string` types.

Comment (beinan) [reply presto-record-decoder/src/main/java/com/facebook/presto/decoder/avro/AvroColumnDecoder.java:206]: I see, thank you for the explanation!

File: presto-record-decoder/src/main/java/com/facebook/presto/decoder/avro/AvroColumnDecoder.java (hunk 2 above)
Comment (yangy0000) [reply presto-record-decoder/src/main/java/com/facebook/presto/decoder/avro/AvroColumnDecoder.java:279]: yes, the value could be Integer, which will fall into 

`type.writeLong(blockBuilder, ((Number) value).longValue());
`
I'm not an expert on this, but my test indicate this line it is compatible with Integer

**Comment by yangy0000** [comment]
> Thank you for this contribution! Looks good to me except a couple clarification questions. Could you also update the release note section in this PR? Thanks!

updated

File: presto-record-decoder/src/main/java/com/facebook/presto/decoder/avro/AvroColumnDecoder.java (hunk 1 above)
Comment (yangy0000) [reply presto-record-decoder/src/main/java/com/facebook/presto/decoder/avro/AvroColumnDecoder.java:21]: @beinan  gentle bump :)

**Review by beinan — APPROVED** [review]
lgtm

**Pull request merged**